nano/node/json_handler.cpp

```diff
@@ -2069,19 +2069,35 @@ void nano::json_handler::database_txn_tracker ()
 void nano::json_handler::delegators ()
 {
 	auto account (account_impl ());
+	auto count (count_optional_impl ());
+	auto threshold (threshold_optional_impl ());
+	auto head_str (request.get_optional<std::string> ("head"));
+
+	nano::account head_account (0);
+	if (head_str)
```

Comment:
Probably shorter variant
```
	if (!ec && head_str.is_initialized ())
	{
		head_account = account_impl (head_str.get ());
	}
```

Reply:
It seems like we have at least 4 places using this logic, probably would make sense to use some separate function then in later PRs...

Reply:
I don't think we can do this because the "account" parameter is already reserved for the representative's address:

```
auto representative (account_impl ());
```

Reply:
```
	if (!ec && start_account_text.is_initialized ())
	{
		start_account = account_impl (start_account_text.get ());
	}
```
In this way it uses `start_account_text` value & not "account" field.
https://github.com/nanocurrency/nano-node/blob/4d915c1fa3f7741299cae041f992dae283319e74/nano/node/json_handler.cpp#L232-L236
Probbaly need different function for such things later...

Reply:
For current variant it still probably need `if (!ec && start_account_text)` to not replace error codes for count or threshold

Reply:
Ah I see now, makes sense, thanks.

>  Probably need different function for such things later...
Agree